packages/app-config-writer/src/preview-config/ui5-yaml.ts

```diff
@@ -222,7 +233,7 @@ export function updatePreviewMiddlewareConfig(
         writeConfig = true;
     }
     //check intent and respect defaults
-    if (intent && `${intent?.object}-${intent?.action}` !== 'app-preview') {
+    if (intent && `${intent?.object}-${intent?.action}` !== defaultIntent) {
```

Comment:
This check will not do what you think it does because `defaultIntent` is an object so it's string will be "[object, object]". If no test complains about this it seems like we need an additional test 🐱